MSAL/test/app/MSALStressTestHelper.m

```diff
@@ -0,0 +1,239 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSALStressTestHelper.h"
+#import "MSALTestAppTelemetryViewController.h"
+#import "MSALTestAppSettings.h"
+#import "NSURL+MSALExtensions.h"
+#import "MSALAuthority.h"
+#import "MSALAccessTokenCacheItem+TestAppUtil.h"
+
+@implementation MSALStressTestHelper
+
+static dispatch_semaphore_t sem = nil;
+static BOOL stop = NO;
+static BOOL runningTest = NO;
+
+#pragma mark - Helpers
+
++ (void)commonSetup
+{
+    [[MSALTestAppTelemetryViewController sharedController] stopTracking];
+    [MSALLogger sharedLogger].level = MSALLogLevelNothing;
+    
+    stop = NO;
+}
+
++ (MSALPublicClientApplication *)createTestPublicApplicationWithLogHandler:(void (^)(NSString *testLogEntry))logHandler
+{
+    MSALTestAppSettings *settings = [MSALTestAppSettings settings];
+    
+    if (![[settings.scopes allObjects] count])
+    {
+        logHandler(@"Please select the scope!");
```

Comment:
These sorts of messages there should be some sort of test utility you can use without having to rely on a passed in block.